stdlib/LinearAlgebra/test/triangular.jl

```diff
@@ -605,6 +605,45 @@ end
     end
 end
 
+@testset "inplace mul of appropriate types should preserve triagular structure" begin
+    for elty1 in (Float32, Float64, ComplexF32, ComplexF64, Int)
+        for elty2 in (Float32, Float64, ComplexF32, ComplexF64, Int)
+            A = UpperTriangular(rand(elty1, 5, 5))
```

Comment:
`rand(Int,...)` may yield very large integers, which may overflow upon multiplication and addition.